e2e/internal/e2e/image.go

```diff
@@ -10,17 +10,10 @@ import (
 	"testing"
 )
 
-var testenv = struct {
-	CmdPath   string `split_words:"true"` // singularity program
-	ImagePath string `split_words:"true"` // base image for tests
-}{}
-
 // EnsureImage checks if e2e test image is already built or built
 // it otherwise.
-func EnsureImage(t *testing.T) {
-	LoadEnv(t, &testenv)
-
-	switch _, err := os.Stat(testenv.ImagePath); {
+func EnsureImage(t *testing.T, env TestEnv) {
```

Comment:
instead of `env` should this be `env.ImagePath`?

Reply:
I guess you are right, but...

This function needs three things:
1. The path to the test image
2. The path to singularity in case the image needs to be built
3. The path to the directory with test data

That's more than half of the information contained in `TestEnv`. In fact I wonder if it should care about `RunDisabled`.

The first one is obviously being used. The second one _should_ be used, but because of the way `RunSingularity` works, it's not obvious that this function depends on that. The last one is tricky, because there's an implicit requirement that the current working directory is one above the directory with the test data.

I can change it to take an image path if you think that's better, but I'm pretty sure it's going to change back to this, or something like this. I thought about making this function a method of TestEnv, but did not implement that.